#!/usr/bin/env python
PACKAGE = "yahboomcar_multi"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("lin_Kp", double_t, 0, "Kp in PID", 1.0, 0, 10)
gen.add("lin_Ki", double_t, 0, "Ki in PID", 0.0, 0, 10)
gen.add("lin_Kd", double_t, 0, "Kd in PID", 1.0, 0, 10)
gen.add("ang_Kp", double_t, 0, "Kp in PID", 0.8, 0, 10)
gen.add("ang_Ki", double_t, 0, "Ki in PID", 0.0, 0, 10)
gen.add("ang_Kd", double_t, 0, "Kd in PID", 1.0, 0, 10)

teams_enum = gen.enum([gen.const("convoyl", int_t, 0, "A small data"),
                       gen.const("vertical", int_t, 1, "A medium data"),
                       gen.const("horizonta", int_t, 2, "A large data")],
                      "An enum to set size")

gen.add("teams", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 2, edit_method=teams_enum)
robot_model_enum = gen.enum([gen.const("omni", int_t, 0, "A medium data"),
                             gen.const("diff", int_t, 1, "A large data")],
                            "An enum to set size")

gen.add("robot_model", int_t, 0, "A size parameter which is edited via an enum", 0, 0, 1, edit_method=robot_model_enum)

gen.add("navigate", bool_t, 0, "navigate in rosbot", False)
gen.add("switch", bool_t, 0, "switch in rosbot", True)
gen.add("dist", double_t, 0, "dist", 0.7, 0.5, 2)
exit(gen.generate(PACKAGE, "RoundListener", "RobotListener"))
